nemo/collections/nlp/models/language_modeling/megatron_gpt_prompt_learning_model.py

```diff
@@ -403,10 +403,13 @@ def test_epoch_end(self, outputs):
 
     def setup_training_data(self, training_data_config=None):
         if self.cfg.data.get('train_ds', None):
+            max_seq_length = self.frozen_model.cfg.encoder_seq_length
+            if "max_seq_length" in self.cfg.data and self.cfg.data.max_seq_length:
+                max_seq_length = self.cfg.data.max_seq_length
             self._train_ds, self._train_dl = self.build_virtual_prompt_dataset(
                 data=self.cfg.data.train_ds,
                 batch_size=self.cfg.global_batch_size,
-                max_seq_length=self.frozen_model.cfg.encoder_seq_length,
+                max_seq_length=max_seq_length,
```

Comment:
Looking at https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/nlp/data/language_modeling/megatron/gpt_prompt_learning_dataset.py#L166

Why might the sample have longer than max_sequence_length even after truncation? IN practice, this might mean that all samples get truncated and none get filtered out.

Reply:
The idea is to remove long examples relative to other examples in training set. So its not really related to max_encoder_seq_length.

Reply:
What I am wondering about is this:

Why might the ' if self.min_seq_length <= len(input_ids) <= self.max_seq_length:' ever be true since we do this truncation step first?

`# Try to truncate input text to fit into the max sequence length
            if len(input_ids) > self.max_seq_length:
                input_ids = self._truncate_input(
                    truncation_field,
                    input_ids,
                    taskname,
                    doc,
                    prompt_template,
                    prompt_template_fields,
                    virtual_token_splits,
                )

            # Skip example if the final length doesn't fit length requirements even after truncation
            if self.min_seq_length <= len(input_ids) <= self.max_seq_length:
                if self.virtual_prompt_source == VirtualPromptSource.PROMPT_ENCODER:
                    taskname_id = self.tokenizer.text_to_ids(taskname)
                elif self.virtual_prompt_source == VirtualPromptSource.NO_PROMPT:
                    taskname_id = -1
                else:
                    raise ValueError("Invalid virtual prompt source specified")

                # Find answer field indices if training and answer_only_loss is True
                answer_start_idx = None
                if answer_only_loss and self.for_train:
                    answer_start_idx = self._find_answer_start(taskname, input_ids, answer_field, doc)

                self.examples.append((taskname_id, input_ids, answer_start_idx))
            else:
                skipped += 1`

Reply:
Yeah it might become redundant. But not handling that in this PR.